### PR TITLE
fix(bar): Fix schema and css for app bar and nav bar

### LINKED ISSUE #1166: [BUG] AppBar and FooterBar shoudl be always visible unless static
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The customization of appbar and footerbar is too loose and not enough standardize. It creates confusion...

### Expected Behavior

Make Appbar and FooterBar always visible unless the map interaction is static. AppBar will hold the about and notification center so it should always be there.

If map is static, we can add/remove the components. If map is dynamic it is always there.

### Steps To Reproduce

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jolevesq**
Closes #1166

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Made modification to the behavior of footer bar, app bar and nav bar. They are now always visible. Reset the array items for app bar and nav bar plus modify some css behavior.

Fixes #1166, #1167

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Modify all configuration to showcase different setup with nav bar, footer bar, app bar

tested all html pages...

https://jolevesq.github.io/geoview

# Checklist:

- [x] I have build __(rush build)__ and deploy __(rush host)__ my PR
- [x] I have connected the issues(s) to this PR
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have created new issue(s) related to the outcome of this PR is needed
-  ~~I have made corresponding changes to the documentation~~
-  ~~I have added tests that prove my fix is effective or that my feature works~~
-  ~~New and existing unit tests pass locally with my changes~~

- - -
This change is [Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169)

### CODE REVIEW

File: packages/geoview-core/src/core/components/app-bar/buttons/geolocator.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import { api, booleanPayload } from '../../../../app';
+import { EVENT_NAMES } from '../../../../api/events/event-types';
+import { IconButton, SearchIcon } from '../../../../ui';
+
+/**
+ * Interface used for geolocator button properties
+ */
+interface GeolocatorProps {
+  className?: string | undefined;
```
Comment (amir-azma) [inline packages/geoview-core/src/core/components/app-bar/buttons/geolocator.tsx:11]: I recommend to have the mandatory props first, then optional props after that. For example in this case you can have mapId first, then after that className as it's optional.

File: packages/geoview-core/src/core/components/geolocator/geolocator.tsx [hunk 2]
```diff
@@ -122,7 +135,7 @@ export function Geolocator() {
   };
 
   return (
-    <Box sx={sxClasses.root} id="geolocator-search">
+    <Box sx={sxClasses.root} visibility={`${active ? 'visible' : 'hidden'}`} id="geolocator-search">
```
Comment (amir-azma) [inline packages/geoview-core/src/core/components/geolocator/geolocator.tsx:138]: Can you try use this code for this line? I don't think we need `$` :
`<Box sx={sxClasses.root} visibility={active ? 'visible' : 'hidden'} id="geolocator-search" />`

**Review by DamonU2 — CHANGES_REQUESTED** [review]
> Reviewed 79 of 79 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169)* status: all files reviewed, 4 unresolved discussions (waiting on @cphelefu, @jolevesq, and @ychoquet)

___
*[`packages/geoview-core/src/core/components/app-bar/app-bar.tsx` line 192 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#-N_ePEet2NKzps735ieG:-N_ePEet2NKzps735ieH:b44rdba) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/app-bar/app-bar.tsx#L192)):*
> ```XML
>             <List className={classes.appBarList}>
>               <ListItem>
>                 <Geolocator className={`${classes.appBarButton}}`} mapId={mapId} />
> ```

I think this can just be {classes.appBarButton}, no need for quotes etc.
___
*[`packages/geoview-core/src/core/components/app-bar/buttons/export.tsx` line 5 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#-N_eO4fo77QqvrHcvkvR:-N_eO4fo77QqvrHcvkvS:b-8lj8sn) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/app-bar/buttons/export.tsx#L5)):*
> ```XML
> 
> /**
>  * Interface used for excport button properties
> ```

*export

File: packages/geoview-core/src/ui/style/theme.tsx [hunk 3]
```diff
@@ -379,8 +379,8 @@ const themeOptions: ThemeOptions = {
     objectFit: 'scale-down',
   },
   appBar: {
-    background: '#1E1E1E',
-    border: '#fff',
+    background: '#FFFFFFee',
```
Comment (cphelefudk) [inline packages/geoview-core/src/ui/style/theme.tsx:382]: minor comment. HTML code code should be 6 characters.

File: packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx [hunk 4]
```diff
@@ -19,14 +19,14 @@ const sxClasses = {
...
     alignItems: 'center',
-    width: '100%',
-    maxHeight: '34px',
-    minHeight: '34px',
+    width: 'calc(100% - 64px)',
+    maxHeight: '35px',
+    minHeight: '35px',
     backdropFilter: 'blur(5px)',
-    backgroundColor: '#000000aa',
+    backgroundColor: '#000000cc',
```
Comment (cphelefudk) [inline packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx:26]: Same here. HTML color should 6 characters. I think this should be '#0000CC'.

File: packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx [hunk 5]
```diff
@@ -19,14 +19,14 @@ const sxClasses = {
...
     pointerEvents: 'all',
     position: 'absolute',
-    left: 0,
```
Comment (cphelefudk) [inline packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx:29]: If you are using absolute this is necessary to define position of the element. Maybe try `left: '64px';`.. or `left:0; marginLeft:64px`

File: packages/geoview-core/src/core/components/geolocator/geolocator.tsx (hunk 2 above)
Comment (cphelefudk) [reply packages/geoview-core/src/core/components/geolocator/geolocator.tsx:138]: Mostly in react they handle `if..else` conditions like this.. `{ active & <Box .../>}`.. if this is if..else for show/hide.

**Review by jolevesq — COMMENTED** [review]
> Reviewed 77 of 79 files at r1.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169)* status: 72 of 79 files reviewed, 7 unresolved discussions (waiting on @amir-azma, @cphelefu, @cphelefudk, @DamonU2, and @ychoquet)

___
*[`packages/geoview-core/src/core/components/app-bar/app-bar.tsx` line 192 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#-N_ePEet2NKzps735ieG:-N_eYlhSDcNOi8IwrikE:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/app-bar/app-bar.tsx#L192)):*
<details><summary><i>Previously, DamonU2 (Damon Ulmi) wrote…</i></summary><blockquote>

I think this can just be {classes.appBarButton}, no need for quotes etc.
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/core/components/app-bar/buttons/export.tsx` line 5 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#-N_eO4fo77QqvrHcvkvR:-N_eWx6nA1I8RQVRgECf:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/app-bar/buttons/export.tsx#L5)):*
<details><summary><i>Previously, DamonU2 (Damon Ulmi) wrote…</i></summary><blockquote>

*export
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/core/components/app-bar/buttons/geolocator.tsx` line 11 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#gh-1267137892:-N_eWyH13y1s2ZEIUu5v:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/app-bar/buttons/geolocator.tsx#L11)):*
<details><summary><i>Previously, amir-azma wrote…</i></summary><blockquote>

I recommend to have the mandatory props first, then optional props after that. For example in this case you can have mapId first, then after that className as it's optional.
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx` line 26 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#gh-1267236739:-N_eX3hi69f-rXXmTJWx:b-u86m2t) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx#L26)):*
<details><summary><i>Previously, cphelefudk wrote…</i></summary><blockquote>

Same here. HTML color should 6 characters. I think this should be '#0000CC'.
</blockquote></details>

The last 2 are opacity
___
*[`packages/geoview-core/src/core/components/geolocator/geolocator.tsx` line 138 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#gh-1267141524:-N_eXLqP9GLIHfIyXX1c:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/geolocator/geolocator.tsx#L138)):*
<details><summary><i>Previously, amir-azma wrote…</i></summary><blockquote>

Can you try use this code for this line? I don't think we need `$` :
`<Box sx={sxClasses.root} visibility={active ? 'visible' : 'hidden'} id="geolocator-search" />`
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/ui/style/theme.tsx` line 382 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#gh-1267233125:-N_eXOy7A3O-wQ5eV1Kq:b-pdngi7) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/ui/style/theme.tsx#L382)):*
<details><summary><i>Previously, cphelefudk wrote…</i></summary><blockquote>

minor comment. HTML code code should be 6 characters. 
</blockquote></details>

Last 2 is for opacity

**Review by jolevesq — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169)* status: 72 of 79 files reviewed, 7 unresolved discussions (waiting on @amir-azma, @cphelefu, @cphelefudk, @DamonU2, and @ychoquet)

___
*[`packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx` line 0 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169#gh-1267238222:-N_eakfvE8UKrRuXUgqJ:bnu1bt9) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/6cc947fdd739175b55ca9ef247f4c7ba593c8756/packages/geoview-core/src/core/components/footer-bar/footer-bar.tsx#L0)):*
<details><summary><i>Previously, cphelefudk wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

If you are using absolute this is necessary to define position of the element. Maybe try `left: '64px';`.. or `left:0; marginLeft:64px`
</blockquote></details>

Set left to 64

**Review by DamonU2 — APPROVED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169)* status: all files reviewed, 5 unresolved discussions (waiting on @amir-azma, @cphelefu, @cphelefudk, and @ychoquet)

**Review by jolevesq — COMMENTED** [review]
> Reviewed 5 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169)* status: all files reviewed, 5 unresolved discussions (waiting on @amir-azma, @cphelefu, @cphelefudk, and @ychoquet)

**Review by jolevesq — COMMENTED** [review]
> Reviewed 2 of 7 files at r2.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1169)* status: all files reviewed, 5 unresolved discussions (waiting on @amir-azma, @cphelefu, @cphelefudk, and @ychoquet)

**Review by amir-azma — APPROVED** [review]
Looks good

**Pull request merged**